detection_rules/rule_validators.py

```diff
@@ -638,6 +677,292 @@ def validate_integration(
+                    log(
+                        f"Warning: Nested multi-field `{field}` found in `{integration}-{stream}`. "
+                        f"Removing parent field from schema for ES|QL validation."
+                    )
+                    utils.delete_nested_key_from_dict(stream_mappings, field_name)
+                utils.combine_dicts(integration_mappings, stream_mappings)
+                index_lookup[f"{integration}-{stream}"] = stream_mappings
+
+        return integration_mappings, index_lookup
+
+    def validate_columns_index_mapping(
+        self, query_columns: list[dict[str, str]], combined_mappings: dict[str, Any]
+    ) -> bool:
+        """Validate that the columns in the ESQL query match the provided mappings."""
+        mismatched_columns: list[str] = []
+
+        for column in query_columns:
+            column_name = column["name"]
+            # Skip Dynamic fields
+            if column_name.startswith(("Esql.", "Esql_priv.")):
+                continue
+            # Skip internal fields
+            if column_name in ("_id", "_version", "_index"):
+                continue
+            column_type = column["type"]
+
+            # Check if the column exists in combined_mappings or a valid field generated from a function or operator
+            keys = column_name.split(".")
+            schema_type = utils.get_column_from_index_mapping_schema(keys, combined_mappings)
+
+            # Validate the type
+            if not schema_type or column_type != schema_type:
+                mismatched_columns.append(
+                    f"Dynamic field `{column_name}` is not correctly mapped. "
+                    f"If not dynamic: expected `{schema_type}`, got `{column_type}`."
+                )
+
+        if mismatched_columns:
+            raise ValueError("Column validation errors:\n" + "\n".join(mismatched_columns))
+
+        return True
+
+    def create_remote_indices(
+        self,
+        elastic_client: Elasticsearch,
+        existing_mappings: dict[str, Any],
+        index_lookup: dict[str, Any],
+        log: Callable[[str], None],
+    ) -> str:
+        """Create remote indices for validation and return the index string."""
+        suffix = str(int(time.time() * 1000))
+        test_index = f"rule-test-index-{suffix}"
+        response = misc.create_index_with_index_mapping(elastic_client, test_index, existing_mappings)
+        log(f"Index `{test_index}` created: {response}")
+        full_index_str = test_index
+
+        # create all integration indices
+        for index, properties in index_lookup.items():
+            ind_index_str = f"test-{index.rstrip('*')}{suffix}"
+            response = misc.create_index_with_index_mapping(elastic_client, ind_index_str, properties)
+            log(f"Index `{ind_index_str}` created: {response}")
+            full_index_str = f"{full_index_str}, {ind_index_str}"
+
+        return full_index_str
+
+    def execute_query_against_indices(
```

Comment:
should this go in remote_validation.py?

Reply:
No, this logical path is highly dependent on the index modifications which are required as part of the query syntax validation. Remote validation _should_ support running one's query against supplied indexes, but since we cannot parse them, we cannot support this as directly, and need to use the syntax validation to help with this.  

Remote validation's purpose is to validate the rule against a stack setup, assuming syntax validation is already done. Given that we cannot separate the two in our cases we are depend on stack validation for syntax validation.